### PR TITLE
Identify failed jobs in lambda and mark as steps=0

### PR DESCRIPTION
**Pull request by Felixoid**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The query to identify failed jobs uses the condition `countIf(conclusion='failure' AND steps <= 5) as killed_jobs`. But we recently changed the jobs, so now it has a lot of steps, like [here](https://api.github.com/repos/ClickHouse/ClickHouse/actions/jobs/19368168635). This change will identify the killed jobs reliably at the receiving stage.

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 13cdb00a67c26ec183179965cd2b74671622ad11 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57600/13cdb00a67c26ec183179965cd2b74671622ad11/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7121719752">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57600/13cdb00a67c26ec183179965cd2b74671622ad11/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: tests/ci/workflow_jobs_lambda/app.py [hunk 1]
```diff
@@ -126,6 +126,20 @@ def send_event_workflow_job(workflow_job: WorkflowJob) -> None:
...
+    if (
+        wf_job.get("status", "") != "completed"
+        or wf_job.get("conclusion", "") != "failure"
+    ):
+        # The task either success or in progress
+        return False
+    return not any(
+        step["name"] == "Complete job" and step["conclusion"] is not None
+        for step in wf_job["steps"]
+    )
```
Comment (mkaynov) [inline tests/ci/workflow_jobs_lambda/app.py:140]: it's a bit hard to follow the logic, maybe
```suggestion
    return any(
        step["name"] == "Complete job" and step["conclusion"] is None
        for step in wf_job["steps"]
    )
```

Comment (mkaynov) [reply tests/ci/workflow_jobs_lambda/app.py:140]: "Complete job" is GH's step and we assume it's always there, right?

File: tests/ci/workflow_jobs_lambda/app.py [hunk 2]
```diff
@@ -141,8 +155,12 @@ def handler(event: dict, context: Any) -> dict:
         logging.error("The event data: %s", event)
         logging.error("The context data: %s", context)
 
-    # We record only finished steps
-    steps = len([step for step in wf_job["steps"] if step["conclusion"] is not None])
+    if killed_job(wf_job):
+        # for killed job we record 0
+        steps = 0
```
Comment (mkaynov) [inline tests/ci/workflow_jobs_lambda/app.py:160]: we use `step == 0` later as a sign that job was killed. correct?
shall we add dedicated "conclusion" value for such jobs?

**Review by maxknv — COMMENTED** [review]
minor commets, questions

File: tests/ci/workflow_jobs_lambda/app.py (hunk 2 above)
Comment (Felixoid) [reply tests/ci/workflow_jobs_lambda/app.py:160]: I'd leave it as it is in the event. FInished failure with step=0 is already a mark. Previously it was `<5`, so now it will be very much the sign

Comment (Felixoid) [reply tests/ci/workflow_jobs_lambda/app.py:160]: Now `steps` is the only thing we mess up. All others are left as is. It's because we don't write the actual structures, but rewrite it to just a number, see line 165+
